test/e2e/suite/conformance/certificates/suite.go

```diff
@@ -400,6 +400,33 @@ func (s *Suite) Define() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 
+		It("should issue a certificate that includes arbitrary key usages", func() {
+			s.checkFeatures(KeyUsagesFeature)
+
+			testCertificate := &cmapi.Certificate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testcert",
+					Namespace: f.Namespace.Name,
+				},
+				Spec: cmapi.CertificateSpec{
+					SecretName: "testcert-tls",
+					DNSNames:   []string{s.newDomain()},
+					IssuerRef:  issuerRef,
+					Usages: []cmapi.KeyUsage{
+						cmapi.UsageSigning,
+						cmapi.UsageDataEncipherment,
+					},
+				},
+			}
+			By("Creating a Certificate")
+			err := f.CRClient.Create(ctx, testCertificate)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for the Certificate to be issued...")
+			err = f.Helper().WaitCertificateIssuedValid(f.Namespace.Name, "testcert", time.Minute*5)
```

Comment:
I'm still slightly nervous that of this area of code as it confuses me 🙈 https://github.com/jetstack/cert-manager/blob/f4f9f7acc47f14a0304cdc78b146fb65cfc0c479/test/e2e/framework/helper/certificates.go#L188-L207

I thought we already had test cases that checked this? Are we sure this didn't slip through due to a dodgy e2e test validation function?

Reply:
I'm pretty certain that it's fine - the issue was the key usages were never set on the CertificateRequest therefore key usages were never set so was always correct

Reply:
How come this test case is passing? https://github.com/jetstack/cert-manager/blob/5312cfc5080fba2f48f5a4493e213ddadd908795/test/e2e/suite/issuers/ca/certificate.go#L161 (which includes: `gen.SetCertificateKeyUsages(v1alpha2.UsageServerAuth, v1alpha2.UsageClientAuth)`)

Reply:
@JoshVanL bump ^

/unassign
/assign @JoshVanL